include/libtorrent/utp_stream.hpp

```diff
@@ -493,6 +518,33 @@ struct TORRENT_EXTRA_EXPORT utp_stream
 		issue_write();
 	}
 
+#if BOOST_VERSION >= 106600
+	// Compatiblity with the async_wait method introduced in boost 1.66
```

Comment:
oh, I see, you added it :)

Reply:
I think this function, along with the `proxy_base` support for `async_wait` is a good candidate for a separate PR (that can land ahead of this big one, to make it just a little bit smaller)